aiohttp/helpers.py

```diff
@@ -852,6 +853,15 @@ def set_cookie(
         self._cookies[name] = value
         c = self._cookies[name]
 
+        if DEBUG:
+            cookie_length = (
+                len(name) + len(value) + sum(len(k) + len(v) for k, v in c.items())
+            )
+            if cookie_length > COOKIE_MAX_LENGTH:
+                server_logger.warning(
```

Comment:
Codecov is now adding inline comments, that's cool. As it suggests, we should add a test for this case. Maybe also a test that our logic matches 4096 bytes in the output.

Something like:
```
input_4096...
assert no_warning_raised
assert len(cookie_header) == 4096

input_4097...
assert warning_raised
assert len(cookie_header) == 4097
```

Reply:
Also, I think this should be `warnings.warn()`, see: https://docs.python.org/3/howto/logging.html#when-to-use-logging